x-pack/plugins/cases/public/components/actions/delete/use_delete_action.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState } from 'react';
+import { EuiIcon, EuiTextColor, useEuiTheme } from '@elastic/eui';
+import { Case } from '../../../../common';
+import { useDeleteCases } from '../../../containers/use_delete_cases';
+
+import * as i18n from './translations';
+import { UseActionProps } from '../types';
+
+const getDeleteActionTitle = (totalCases: number): string =>
+  totalCases > 1 ? i18n.BULK_ACTION_DELETE_LABEL : i18n.DELETE_ACTION_LABEL;
+
+export const useDeleteAction = ({ onAction, onActionSuccess, isDisabled }: UseActionProps) => {
+  const euiTheme = useEuiTheme();
+  const [isModalVisible, setIsModalVisible] = useState<boolean>(false);
+  const [caseToBeDeleted, setCaseToBeDeleted] = useState<Case[]>([]);
+  const onCloseModal = useCallback(() => setIsModalVisible(false), []);
+  const openModal = useCallback(
+    (selectedCases: Case[]) => {
+      onAction();
+      setIsModalVisible(true);
+      setCaseToBeDeleted(selectedCases);
+    },
+    [onAction]
+  );
+
+  const { mutate: deleteCases } = useDeleteCases();
+
+  const onConfirmDeletion = useCallback(() => {
+    onCloseModal();
+    deleteCases(
+      {
+        caseIds: caseToBeDeleted.map(({ id }) => id),
+        successToasterTitle: i18n.DELETED_CASES(caseToBeDeleted.length),
+      },
+      { onSuccess: onActionSuccess }
+    );
+  }, [deleteCases, onActionSuccess, onCloseModal, caseToBeDeleted]);
+
+  const color = isDisabled ? euiTheme.euiTheme.colors.disabled : 'danger';
+
+  const getAction = (selectedCases: Case[]) => {
+    return {
+      name: <EuiTextColor color={color}>{getDeleteActionTitle(selectedCases.length)}</EuiTextColor>,
+      onClick: () => openModal(selectedCases),
+      disabled: isDisabled,
```

Comment:
I agree with you. We should hide the "bulk" options if the user has not selected any cases.

Reply:
I am thinking of leaving the prop in case we need it in the future. I think it is nice for consumers to be able to disable the actions if needed. What do you think?